mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/repositories/VirksomhetRepository.kt

```diff
@@ -0,0 +1,131 @@
+package no.nav.mulighetsrommet.api.repositories
+
+import kotliquery.Row
+import kotliquery.queryOf
+import no.nav.mulighetsrommet.api.domain.dto.VirksomhetDto
+import no.nav.mulighetsrommet.api.utils.*
+import no.nav.mulighetsrommet.database.Database
+import no.nav.mulighetsrommet.database.utils.QueryResult
+import no.nav.mulighetsrommet.database.utils.query
+import org.intellij.lang.annotations.Language
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class VirksomhetRepository(private val db: Database) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    fun upsert(virksomhetDto: VirksomhetDto): QueryResult<Unit> = query {
+        logger.info("Lagrer virksomhet ${virksomhetDto.organisasjonsnummer}")
+        if (virksomhetDto.overordnetEnhet != null) {
+            throw IllegalArgumentException("Virksomhet må være en overordnet enhet")
+        }
```

Comment:
Usikker på om det blir riktig at vi bare skal ha overordnede enheter i tabellen i databasen. Vi har funksjonalitet i frontend for å velge underenheter også. Per nå hentes de via REST-kall mot Brreg, men vi ønsker kanskje å gå via db her også kan jeg tenke meg?

Reply:
Ah, ser lenger nede at du har to tabeller ja, en for virksomhet og en for underenhet.